fdbclient/FileBackupAgent.actor.cpp

```diff
@@ -576,21 +576,21 @@ struct EncryptedRangeFileWriter : public IRangeFileWriter {
 		wPtr = mutateString(buffer);
 	}
 
-	static void validateEncryptionHeader(Optional<Reference<BlobCipherKey>> headerCipherKey,
+	static void validateEncryptionHeader(Reference<BlobCipherKey> headerCipherKey,
 	                                     Reference<BlobCipherKey> textCipherKey,
 	                                     Optional<BlobCipherDetails> headerCipherDetails,
 	                                     BlobCipherDetails textCipherDetails) {
 		// Validate encryption header 'cipherHeader' details
-		if (headerCipherKey.present() && headerCipherKey.get().isValid() && headerCipherDetails.present() &&
-		    !(headerCipherDetails.get().baseCipherId == headerCipherKey.get()->getBaseCipherId() &&
-		      headerCipherDetails.get().encryptDomainId == headerCipherKey.get()->getDomainId() &&
-		      headerCipherDetails.get().salt == headerCipherKey.get()->getSalt())) {
+		if (headerCipherKey.isValid() && headerCipherDetails.present() &&
```

Comment:
nit: may be add a helper method in BlobCipherDetail to compare when `BlobCipherKey` is passed

Reply:
from older commit